src/explorer/webview.ts

```diff
@@ -137,59 +147,46 @@ class ExplorerWebview {
     /**
      * Cleans up the result details section, removing event handlers and elements
      */
-    private cleanUpResultDetails() {
+    private cleanUpResultDetails(): void {
         // Remove event handlers
-        const tabContainer = document.getElementById("tabcontainer");
-        if (tabContainer !== null) {
-            for (let i = 0; i < tabContainer.children.length; i++) {
-                tabContainer.children.item(i).removeEventListener("click", this.onTabClickedBind);
+        const tabContainer: HTMLDivElement | undefined = getOptionalDocumentElementById(document, "tabcontainer", HTMLDivElement);
```

Comment:
What is the intent of down casting to `HTMLDivElement` here?

Reply:
The idea is to have type-script code actually ensure that element type you are working on is really what you think it is. But yes, in the same regards, this code should be using the least contract it needs to do it's work.

Changing to HTMLElement.

---
In reply to: [397699182](https://github.com/microsoft/sarif-vscode-extension/pull/211#discussion_r397699182) [](ancestors = 397699182)